app/controllers/index.rb

```diff
@@ -1,3 +1,16 @@
 get '/' do
   erb :index
 end
+
+get '/register.json' do
+  erb(:_register, :layout => false)
+end
+
+post '/register.json' do
+  content_type :json
+  if params['email'].match('.+@{1}.+\..+')
```

Comment:
If you are returning an HTML snippet, I dont think it needs to be wrapped as a JSON object.